packages/core/src/probes/types/component-tree-node/merged-component-tree-node.interface.ts

```diff
@@ -0,0 +1,7 @@
+import { ComponentTreeChange } from './component-tree-change.type';
+import { ComponentTreeNode } from './component-tree-node.interface';
+
+export interface MergedComponentTreeNode extends ComponentTreeNode {
```

Comment:
Is this a `composite`? If so, can we name as such?

Also can we add some documentation (e.g. jsdoc comments to describe the purpose of this type (and possibly other related types)? There's already a `ComponentTreeNode`, so a very brief description of the intention would be ideal. On that note, some comments on some functions/types that are not completely self explanatory or could be viewed as subjective would help a bit in some cases to clarify intent.

Reply:
it's not a composite actually. we could add ? members to component tree node interface instead, but i don't like doing that too much.